__tests__/event_manager.test.ts

```diff
@@ -16,20 +16,72 @@ describe('Event Handler', function() {
     ain.em.disconnect();
   });
 
-  it('Subscribe to BLOCK_FINALIZED', (done) => {
-    ain.em.subscribe('BLOCK_FINALIZED', {
-      block_number: null,
-    }, (data) => {
-      done();
+  describe('BLOCK_FINALIZED', () => {
+    it('Subscribe to BLOCK_FINALIZED', (done) => {
+      ain.em.subscribe('BLOCK_FINALIZED', {
+        block_number: null,
+      }, (data) => {
+        done();
+      });
+    });
+
+    it('Subscribe to BLOCK_FINALIZED with wrong config', (done) => {
+      ain.em.subscribe('BLOCK_FINALIZED', {
+        block_number: -1,
+      }, (data) => {
+      }, (err) => {
+        done();
+      });
     });
   });
 
-  it('Subscribe to BLOCK_FINALIZED with wrong config', (done) => {
-    ain.em.subscribe('BLOCK_FINALIZED', {
-      block_number: -1,
-    }, (data) => {
-    }, (err) => {
-      done();
+  describe('VALUE_CHANGED', () => {
+    const [ testAccount ] = ain.wallet.create(1);
+    const testAppName = `test_${Date.now()}`;
+    const testAppPath = `/apps/${testAppName}`;
+
+    beforeAll(async () => {
+      // Create test app
+      ain.wallet.setDefaultAccount(testAccount);
+      await ain.db.ref(`/manage_app/${testAppName}/create/${Date.now()}`).setValue({
+        value: { admin: { [testAccount]: true } },
+      });
+    });
+
+    it('Subscribe to VALUE_CHANGED', (done) => {
+      let blockEventCheck = false;
+      let userEventCheck = false;
+
+      ain.em.subscribe('VALUE_CHANGED', {
+        path: testAppPath,
+        event_source: null,
+      }, (event) => {
+        switch (event.event_source) {
+          case 'BLOCK':
+            blockEventCheck = true;
+            break;
+          case 'USER':
+            userEventCheck = true;
+            break;
+        }
+        if (blockEventCheck && userEventCheck) {
```

Comment:
could you use expect() or assert() instead?

Reply:
In this case we can't use expect because we need to test that all events are coming. Instead, expect is added to the two newly added test cases!

Reply:
When the first event came in, `blockEventCheck && userEventCheck` became false, so the test ended. Perhaps you mean the code below?

```
        expect(blockEventCheck && userEventCheck).toBe(true);
        if (blockEventCheck && userEventCheck) {
          done();
        }
```

Reply:
To cover that case, I think the number of the events needs to be counted?

Reply:
hmm, I'm confused. Can you show me an example??

Reply:
something like below?
```
if (blockEventCount + userEventCount === 2) {
  expect(blockEventCount).toBe(1);
  expect(userEventCount).toBe(1);
  done();
}
```

Reply:
It seems impossible because `userEventCount` can be greater than 1! `USER` events are propagated not only when a transaction is directly received, but also when executing a transaction in the pool. (ref: https://github.com/ainblockchain/ain-blockchain/blob/develop/node/index.js#L1068)

Reply:
Isn't that redundant case, which can be avoided?

Reply:
My question is "should we emit user events in such cases?"

Reply:
I defined the above situation as a `USER` event in perspective of executing stacked transactions sent by the user. So, I defined as `USER` event, except for temporary execution like tempDB. I think it's up to we to define it, what do you think?

Reply:
yes, exactly. why do we need such redundant events.

Reply:
In this case, as the base database changes, it will be different from the before value of the first USER event, shouldn't the user be notified that the default database has changed?

Reply:
Discussed offline) l change `USER` event to be executed only when a transaction is received from JSON_RPC.

Reply:
[Related PR](https://github.com/ainblockchain/ain-blockchain/pull/1036) was opened and test code logic was changed to use blockEventCount and userEventCount.